lib/msf/java/rmi/builder.rb

```diff
@@ -0,0 +1,96 @@
+# -*- coding: binary -*-
+
+require 'rex/java/serialization'
+
+module Msf
+  module Java
+    module Rmi
+      module Builder
+        # Builds a RMI header stream
+        #
+        # @param opts [Hash{Symbol => <String, Fixnum>}]
+        # @option opts [String] :signature
+        # @option opts [Fixnum] :version
+        # @option opts [Fixnum] :protocol
+        # @return [Rex::Proto::Rmi::Model::OutputHeader]
+        def build_header(opts = {})
+          signature = opts[:signature] || Rex::Proto::Rmi::Model::SIGNATURE
+          version = opts[:version] || 2
+          protocol = opts[:protocol] || Rex::Proto::Rmi::Model::STREAM_PROTOCOL
+
+          header = Rex::Proto::Rmi::Model::OutputHeader.new(
+              signature: signature,
+              version: version,
+              protocol: protocol)
+
+          header
+        end
+
+        # Builds a RMI call stream
+        #
+        # @param opts [Hash{Symbol => <Fixnum, Array>}]
+        # @option opts [Fixnum] :message_id
+        # @option opts [Fixnum] :object_number Random to identify the object.
+        # @option opts [Fixnum] :uid_number Identifies the VM where the object was generated.
+        # @option opts [Fixnum] :uid_time Time where the object was generated.
+        # @option opts [Fixnum] :uid_count Identifies different instance of the same object generated from the same VM
+        #   at the same time.
+        # @option opts [Fixnum] :operation On JDK 1.1 stub protocol the operation index in the interface. On JDK 1.2
+        #   it is -1.
+        # @option opts [Fixnum] :hash On JDK 1.1 stub protocol the stub's interface hash. On JDK1.2 is a hash
+        #   representing the method to call.
+        # @option opts [Array] :arguments
+        # @return [Rex::Proto::Rmi::Model::Call]
+        def build_call(opts = {})
```

Comment:
This is another example where I think keyword arguments are nicer than an option hash.